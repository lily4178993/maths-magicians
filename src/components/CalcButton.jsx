import React from 'react';
import PropTypes from 'prop-types';
import { highlightedKeys } from './data';

const CalcButton = ({ symbol, buttonEvent }) => {
  const isButtonOrange = highlightedKeys.includes(symbol);

  const handleClick = (e) => {
    buttonEvent(e.target.textContent);
  };

  return (
    <button
      type="button"
      title={symbol}
      onClick={handleClick}
      className={isButtonOrange ? 'orange' : null}
    >
      {symbol}
    </button>
  );
};

CalcButton.propTypes = {
  symbol: PropTypes.string.isRequired,
  buttonEvent: PropTypes.func.isRequired,
};
export default CalcButton;
