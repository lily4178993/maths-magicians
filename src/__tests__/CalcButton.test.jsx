import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import CalcButton from '../components/CalcButton';

test('PropTypes snapshot', () => {
  const buttonEvent = jest.fn();
  const { asFragment } = render(<CalcButton symbol="+" buttonEvent={buttonEvent} />);
  expect(asFragment()).toMatchSnapshot();
});

test('Simulate user interaction', () => {
  const buttonEvent = jest.fn();
  const { getByText } = render(<CalcButton symbol="+" buttonEvent={buttonEvent} />);
  const buttonElement = getByText('+');

  fireEvent.click(buttonElement);

  expect(buttonEvent).toHaveBeenCalledWith('+');
});
