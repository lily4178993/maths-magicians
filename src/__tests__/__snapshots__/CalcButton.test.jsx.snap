// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropTypes snapshot 1`] = `
<DocumentFragment>
  <button
    class="orange"
    title="+"
    type="button"
  >
    +
  </button>
</DocumentFragment>
`;
